pkg/kademlia/dht.go

```diff
@@ -5,37 +5,178 @@ package kademlia
 
 import (
 	"context"
+	"crypto/rand"
+	"fmt"
+	"strings"
 
-	"storj.io/storj/protos/overlay"
+	bkad "github.com/coyle/kademlia"
+	"github.com/zeebo/errs"
+
+	proto "storj.io/storj/protos/overlay"
 )
 
+// NodeErr is the class for all errros petaining to node operations
+var NodeErr = errs.Class("node error")
+
+//TODO: shouldn't default to TCP but not sure what to do yet
+var defaultTransport = proto.NodeTransport_TCP
+
 // Kademlia is an implementation of kademlia adhering to the DHT interface.
 type Kademlia struct {
+	rt             RoutingTable
+	bootstrapNodes []proto.Node
+	ip             string
+	port           string
+	stun           bool
+	dht            *bkad.DHT
+}
+
+// NewKademlia returns a newly configured Kademlia instance
+func NewKademlia(bootstrapNodes []proto.Node, ip string, port string, stun bool) Kademlia {
```

Comment:
if we're allowing ourselves to diverge from the other lib, i don't think it makes much since to have stun be a flag here. we should definitely always try and do nat traversal if necessary

Reply:
OK! I wasn't sure if there was a reason you would/wouldn't want to use stun so I left it there. Will remove.